roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,104 @@
+
+        pulled_images = self.pull_images(self.add_image_tags(required_images, openshift_release), tmp, task_vars)
+
+        if "failed_images" in pulled_images:
+            failed_pulls.update(pulled_images["failed_images"])
+
+        if not openshift_image_tag:
+            # if we did not fail to pull at least one image, then
+            # we have successfully changed the host environment
+            if len(failed_pulls) < len(required_images):
+                changed = True
+            return {"changed": changed, "failed": True, "msg": "Unable to retrieve fully qualified image tag. Failed to fetch images: %s" % (set(failed_pulls).update(self.qualified_docker_images(is_origin_deployment)))}
+
+        # pull each required image that requires a qualified image
+        # tag (v0.0.0.0) rather than a standard release format tag (v0.0).
+        # We want to do this check for both containerized and
+        # non-containerized installations.
+        pulled_qualified_images = self.pull_images(self.add_image_tags(self.qualified_docker_images(is_origin_deployment), "v" + openshift_image_tag), tmp, task_vars)
+        if "failed_images" in pulled_qualified_images:
+            failed_pulls.update(pulled_qualified_images["failed_images"])
+
+        if failed_pulls:
+            if len(failed_pulls) < len(required_images) + len(self.qualified_docker_images(is_origin_deployment)):
+                changed = True
+            return {"changed": changed, "failed": True, "msg": "Failed to pull required docker images: %s" % failed_pulls}
+
+        return {"changed": True}
+
+    @staticmethod
+    def containerized_docker_images(is_origin_deployment):
+        if is_origin_deployment:
+            return [
+                "openshift/origin"
+            ]
+        return [
+            "openshift3/ose"
+        ]
+
+    @staticmethod
+    def rpm_docker_images(is_origin_deployment):
+        return [
+            "openshift3/registry-console",
+        ]
+
+    @staticmethod
+    def qualified_docker_images(is_origin_deployment):
+        if is_origin_deployment:
+            return [
+                "openshift/origin-haproxy-router"
+                "openshift/origin-docker-registry"
+                "openshift/origin-deployer"
+                "openshift/origin-pod"
+            ]
+        return [
+            "openshift3/ose-haproxy-router",
+            "openshift3/ose-docker-registry",
+            "openshift3/ose-deployer",
+            "openshift3/ose-pod",
+        ]
+
+    def pull_images(self, tagged_images, tmp, task_vars):
+        failed = set()
+        for image in tagged_images:
+            args = {
+                "name": image
+            }
+
+            res = self.module_executor("docker_image", args, tmp, task_vars)
+            if "failed" in res and res["failed"]:
+                failed.add(args["name"])
+
+        if failed:
+            return {"failed": True,  "failed_images": failed}
```

Comment:
The logic for setting `changed` could be here, instead of repeated in the callers of `pull_images`:
- https://github.com/openshift/openshift-ansible/pull/3461/files#diff-f6d4c415edd5332159aa1c77eb72b757R35
- https://github.com/openshift/openshift-ansible/pull/3461/files#diff-f6d4c415edd5332159aa1c77eb72b757R48